src/analyze_fs.c

```diff
@@ -387,14 +387,17 @@ struct manifest *full_manifest_from_directory(int version)
 {
 	struct manifest *manifest;
 	char *dir;
+	int numthreads = getenv("SWUPD_NUM_THREADS") ?
+		atoi(getenv("SWUPD_NUM_THREADS")) :
+		sysconf(_SC_NPROCESSORS_ONLN);
```

Comment:
I agree, a function would be better. But how would error handling/reporting work? Just exit()?

Is there enough interest in this enhancement to spend more time on this? I only used it while debugging, and now could drop the entire patch again, too.